### PR TITLE
fix bad use of static builder method

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis error in ModuleUtils found by static analysis looks like a potentially serious bug

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/AppEngineDescriptor.java [hunk 1]
```diff
@@ -71,7 +76,7 @@ public String getProjectVersion() throws CoreException {
   }
 
   /**
-   * @return service ID parsed from the &lt;application&gt; element of the appengine-web.xml, or
+   * @return service ID parsed from the &lt;service&gt; element of the appengine-web.xml, or
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/AppEngineDescriptor.java]: oops, good catch.  It should also have `<module>` for backwards compatibility.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class ModuleUtilsTest {
+
+  @Before
+  public void turnOffLogging() {
+    Logger logger = Logger.getLogger(ModuleUtils.class.getName());
+    logger.setLevel(Level.OFF);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtilsTest.java]: We should restore the logging level, no?

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtils.java [hunk 3]
```diff
@@ -35,22 +34,22 @@
   private static final Logger logger = Logger.getLogger(ModuleUtils.class.getName());
 
   /**
-   * Retrieve the &lt;service&gt; or &lt;module&gt; identifier from the <tt>appengine-web.xml</tt>.
-   * If an identifier is not found, or <tt>appengine-web.xml</tt> is not found, then returns
-   * "default".
+   * Retrieve the &lt;service&gt; or &lt;module&gt; identifier from <tt>appengine-web.xml</tt>.
+   * If an identifier is not found, then returns "default".
+   * If <tt>appengine-web.xml</tt> is not found, then returns null.
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtils.java]: The original implementation was wrong, it is true.  My intent was for this method to be `@NonNull`.

But my choice here, to return _"default"_ if the file didn't exist, was deliberate: if you want the untrammelled information, use `AppEngineDescriptor` directly.  In practice our `canModify()` first verifies all modules have an App Engine facet, which should ensure they have an `appengine-web.xml`.

If you really think we should return `null` here, then we need to fix up `LocalAppEngineServerDelegate` too.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtilsTest.java (hunk 2 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtilsTest.java]: We can, though honestly in a test suite I'm tempted to leave it off. zAs is, it just clutters the logs.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtils.java (hunk 3 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtils.java]: I don't think we should return null, but that's what I noticed the current implementation doing and I wasn't sure if that was a mistake or not. I'll update it.

File: plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/AppEngineDescriptor.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/AppEngineDescriptor.java]: It does look at the module elements as a backup.

**Pull request merged**